Content.Shared/Burial/BurialSystem.cs

```diff
@@ -0,0 +1,168 @@
+    {
+        if (args.Handled)
+            return;
+
+        // allows someone to help you dig yourself out
+        if (TryComp<ActiveGraveComponent>(uid, out var active) && !active.DiggingSelfOut)
+            return;
+
+        if (TryComp<ShovelComponent>(args.Used, out var shovel))
+        {
+            _doAfterSystem.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, component.DigDelay / shovel.SpeedModifier, new GraveDiggingDoAfterEvent(), uid, target: args.Target, used: uid)
+            {
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                NeedHand = true,
+                BreakOnHandChange = true
+            });
+
+            StartDigging(uid, args.User, args.Used, component);
+        }
+        else
+        {
+            _popupSystem.PopupClient(Loc.GetString("grave-digging-requires-tool", ("grave", args.Target)), uid, args.User);
+        }
+    }
+
+    private void OnAfterInteractUsing(EntityUid uid, GraveComponent component, AfterInteractUsingEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        // don't place shovels on the grave, only dig
+        if (HasComp<ShovelComponent>(args.Used))
+            args.Handled = true;
+    }
+
+    private void OnActivate(EntityUid uid, GraveComponent component, ActivateInWorldEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        _popupSystem.PopupClient(Loc.GetString("grave-digging-requires-tool", ("grave", args.Target)), uid, args.User);
+    }
+
+    private void OnGraveDigging(EntityUid uid, GraveComponent component, GraveDiggingDoAfterEvent args)
+    {
+        RemComp<ActiveGraveComponent>(uid);
+
+        if (args.Cancelled || args.Handled)
+            return;
+
+        component.DiggingComplete = true;
+
+        if (args.Used != null)
+            _storageSystem.ToggleOpen(args.User, uid);
+        else
+            _storageSystem.TryOpenStorage(args.User, uid); //can only claw out
+    }
+
+    private void StartDigging(EntityUid uid, EntityUid user, EntityUid? used, GraveComponent component)
+    {
+        ActiveGraveComponent activeComp = EnsureComp<ActiveGraveComponent>(uid);
+
+        if (used != null)
+        {
+            _popupSystem.PopupClient(Loc.GetString("grave-start-digging-user", ("grave", uid), ("tool", used)), user, user);
+            _popupSystem.PopupEntity(Loc.GetString("grave-start-digging-others", ("user", user), ("grave", uid), ("tool", used)), user, Filter.PvsExcept(user), true);
+            if (_netManager.IsServer)
+                activeComp.Stream = _audioSystem.PlayPvs(component.DigSound, uid).Value.Entity;
+        }
+        else
+        {
+            activeComp.DiggingSelfOut = true;
+            _popupSystem.PopupClient(Loc.GetString("grave-start-digging-user-trapped", ("grave", uid)), user, user, PopupType.Medium);
+        }
+    }
```

Comment:
Why is IsServer used here and it's not predicted?

Reply:
I had been struggling to handle audio loops in shared, but I think I've got it working with prediction now.